### PR TITLE
ING-787/ING-788: Improve http handling.

### PR DESCRIPTION
**Pull request by brett19**

### CODE REVIEW

**Review by avsej — APPROVED** [review]

File: agent.go [hunk 1]
```diff
@@ -543,9 +543,9 @@ func makeHTTPTransport(tlsConfig *tls.Config) *http.Transport {
 			tlsConn := tls.Client(tcpConn, tlsConfig)
 			return tlsConn, nil
 		},
-		// MaxIdleConns:        maxIdleConns,
-		// MaxIdleConnsPerHost: maxIdleConnsPerHost,
-		// IdleConnTimeout:     idleTimeout,
+		MaxIdleConns:        0,
```
Comment (chvck) [inline agent.go:546]: Real nitty nit: A comment here explaining what 0 means.

File: cbhttpx/errors.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package cbhttpx
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrConnectError = errors.New("http connect failed")
```
Comment (chvck) [inline cbhttpx/errors.go:9]: ErrConnect? The Error suffix seems superfluous.

File: cbhttpx/client.go [hunk 3]
```diff
@@ -36,6 +37,15 @@ func (c Client) GetHttpClient() *http.Client {
 func (c Client) Do(req *http.Request) (*http.Response, error) {
 	resp, err := c.GetHttpClient().Do(req)
 	if err != nil {
+		var netOpError *net.OpError
+		if errors.As(err, &netOpError) {
+			if netOpError.Op == "dial" {
```
[the remark this answers is not in the document]
Comment (brett19) [reply cbhttpx/client.go:42]: Unfortunately not.

File: agent.go (hunk 1 above)
Comment (brett19) [reply agent.go:546]: MaxIdleConbns itself documents that.

**Pull request merged**